lib/profiles.js

```diff
@@ -0,0 +1,319 @@
+ *
+ * @param {String} accountId the id of the account to associate with the
+ *                           Profile.
+ * @param {Object} settings the settings for a Profile.
+ *
+ * @return {Promise<Profile>} resolves to a profile's settings.
+ */
+exports.create = async ({actor, accountId, settings} = {}) => {
+  const {profileAgent} = await profileAgents.create({accountId});
+  // 1. Use the ProfileAgent to generate a new AsymmetricKey in its  keystore.
+  //    This key will be the Profile's ZcapKey.
+  const {capabilityAgent: paZcapAgent} = profileAgent;
+  const profileKeystore = await kms.createKeystore(
+    {capabilityAgent: paZcapAgent, referenceId: 'primary'});
+  const profileKeystoreAgent = kms.getKeystoreAgent(
+    {capabilityAgent: paZcapAgent, keystore: profileKeystore});
+  const key = await profileKeystoreAgent.generateKey(
+    {type: 'Ed25519VerificationKey2018', kmsModule: KMS_MODULE});
+  // 2. Use the ProfileZcapKey to generate the Profile DID
+  // FIXME: Should be setting the profile DID to a V1 DID
+  await utils.setKeyId({key, method: 'key'});
+  // 3. Delegate a capability from the Profile DID to the ProfileAgent to
+  //    allow the ProfileAgent to sign with the ProfileZcapKey for the purpose
+  //    of performing capability invocations.
+  const delegateZcapKeyRequest = {
+    referenceId: `${key.id}-key-capabilityInvocation`,
+    // string should match KMS ops
+    allowedAction: 'sign',
+    controller: paZcapAgent.id,
+    invocationTarget: {
+      id: (await key.getKeyDescription()).id,
+      type: 'Ed25519VerificationKey2018',
+      proofPurpose: 'capabilityInvocation',
+      verificationMethod: key.id
+    }
+  };
+  const zcap = await delegateCapability(
+    {signer: key, request: delegateZcapKeyRequest});
+  // 4. Save the delegated capability in the profile agent's capability set
+  const capabilitySet = {
+    sequence: 0,
+    profileAgent: profileAgent.id,
+    zcaps: [zcap]
+  };
+  await capabilitySets.create({capabilitySet});
+  // 5. Update the `controller` for the Profile Keystore, change from
+  // `profileAgent.id` to Profile DID.
+  // FIXME: Need to properly find the DID associated with the key
+  const profileDid = key.id.split('#')[0];
+  const updatedKeystore = await kms.updateKeystoreController(
+    {id: profileKeystore.id, controller: profileDid});
+  const {httpsAgent} = brHttpsAgent;
+  const kmsClient = new KmsClient({keystore: updatedKeystore.id, httpsAgent});
+  const invocationSigner = new AsymmetricKey({
+    capability: zcap,
+    invocationSigner: paZcapAgent.getSigner(),
+    kmsClient
+  });
+  await profileAgents.update({
+    profileAgent: {
+      ...profileAgent,
+      sequence: profileAgent.sequence + 1,
+      profile: profileDid,
+      profileZcapId: zcap.id
+    }
+  });
+  // 6. Store a single 'user' object in the profile's 'users' EDV
+  // TODO: Make sure these config variables works in production mode
+  const usersReferenceId = edvs.getReferenceId('users');
```

Comment:
I think it's becoming more clear now that the difference between a `shared` profile and one that isn't is that we don't need a `users` EDV for a non-shared profile. That may/may not imply we can do all the `users` stuff on the frontend to "enable" sharing by creating a `users` EDV and adding the root admin/manager profile agent to it as the first user.